Dockerfile

```diff
@@ -1,15 +1,50 @@
 ARG GO_VERSION=1.20
-ARG BUILD_TYPE=
 FROM golang:$GO_VERSION
+
+ARG BUILD_TYPE=
+ARG GO_TAGS=
+ARG CUDA_MAJOR_VERSION=11
+ARG CUDA_MINOR_VERSION=7
+
+ENV BUILD_TYPE=${BUILD_TYPE}
+ENV GO_TAGS=${GO_TAGS}
+ENV NVIDIA_DRIVER_CAPABILITIES=compute,utility
+ENV NVIDIA_REQUIRE_CUDA="cuda>=${CUDA_MAJOR_VERSION}.0"
+ENV NVIDIA_VISIBLE_DEVICES=all
+ENV HEALTHCHECK_ENDPOINT=http://localhost:8080/readyz
 ENV REBUILD=true
+
 WORKDIR /build
-RUN apt-get update && apt-get install -y cmake curl libgomp1 libopenblas-dev libopenblas-base libopencv-dev libopencv-core-dev libopencv-core4.5 ca-certificates
+
+RUN apt-get update && \
+    apt-get install -y ca-certificates cmake curl
+
+RUN if [ "${BUILD_TYPE}" = "cublas" ]; then \
+    apt-get install -y software-properties-common && \
+    apt-add-repository contrib && \
+    curl -O https://developer.download.nvidia.com/compute/cuda/repos/debian11/x86_64/cuda-keyring_1.0-1_all.deb && \
+    dpkg -i cuda-keyring_1.0-1_all.deb && \
+    rm -f cuda-keyring_1.0-1_all.deb && \
+    apt-get update && \
+    apt-get install -y cuda-nvcc-${CUDA_MAJOR_VERSION}-${CUDA_MINOR_VERSION} libcublas-dev-${CUDA_MAJOR_VERSION}-${CUDA_MINOR_VERSION} \
+    ; fi
+ENV PATH /usr/local/cuda/bin:${PATH}
+
+RUN if [ "${BUILD_TYPE}" = "openblas" ]; then \
+    apt-get install -y libopenblas-dev \
+    ; fi
+
+RUN if [ "${GO_TAGS}" = "stablediffusion" ]; then \
+    apt-get install -y libopencv-dev && \
```

Comment:
This is looking good!

Can we keep opencv and openblas in the default images? I'm more thorn to enable them by default in the long-run rather then having them disabled.

Only opencv seems bringing up more deps in the image, but I think that deserve its own discussion

Reply:
Sure. The idea was first to isolate what libraries are needed for what feature. This is the kind of info that is easily lost.

I can add some build options to force some librairies installation even if there are not needed?

Reply:
gotcha, makes totally sense and sounds good to me to keep the information around. fine for me to force re-installation, or just even leaving a comment

Reply:
I've just added some comments